src/tests/suite/daos_container.c

```diff
@@ -955,37 +972,628 @@ expect_cont_open_access(test_arg_t *arg0, uint64_t perms, uint64_t flags,
+	int		rc;
+
+	rc = test_setup((void **)&arg, SETUP_EQ, arg0->multi_rank,
+			DEFAULT_POOL_SIZE, NULL);
+	assert_int_equal(rc, 0);
+
+	print_message("Not asking for any props\n");
+	expect_co_query_access(arg, NULL,
+			       DAOS_ACL_PERM_CONT_ALL &
+			       ~DAOS_ACL_PERM_GET_PROP &
+			       ~DAOS_ACL_PERM_GET_ACL,
+			       -0);
+
+	print_message("All props with no get-prop access\n");
+	prop = get_query_prop_all();
+	expect_co_query_access(arg, prop,
+			       DAOS_ACL_PERM_CONT_ALL & ~DAOS_ACL_PERM_GET_PROP,
+			       -DER_NO_PERM);
+	daos_prop_free(prop);
+
+	print_message("All props with no get-ACL access\n");
+	prop = get_query_prop_all();
+	expect_co_query_access(arg, prop,
+			       DAOS_ACL_PERM_CONT_ALL & ~DAOS_ACL_PERM_GET_ACL,
+			       -DER_NO_PERM);
+	daos_prop_free(prop);
+
+	print_message("All props with only prop and ACL access\n");
+	prop = get_query_prop_all();
+	expect_co_query_access(arg, prop,
+			       DAOS_ACL_PERM_GET_PROP | DAOS_ACL_PERM_GET_ACL,
+			       0);
+	daos_prop_free(prop);
+
+	/*
+	 * ACL props can only be accessed by users with get-ACL permission
+	 */
+	print_message("ACL prop with no get-ACL access\n");
+	prop = get_single_query_prop(DAOS_PROP_CO_ACL);
+	expect_co_query_access(arg, prop,
+			       DAOS_ACL_PERM_CONT_ALL & ~DAOS_ACL_PERM_GET_ACL,
+			       -DER_NO_PERM);
+	daos_prop_free(prop);
+
+	print_message("ACL prop with only get-ACL access\n");
+	prop = get_single_query_prop(DAOS_PROP_CO_ACL);
+	expect_co_query_access(arg, prop,
+			       DAOS_ACL_PERM_GET_ACL,
+			       0);
+	daos_prop_free(prop);
+
+	/*
+	 * Props unrelated to access/ACLs can only be accessed by users with
+	 * the get-prop permission
+	 */
+	print_message("Non-access-related prop with no get-prop access\n");
+	prop = get_single_query_prop(DAOS_PROP_CO_LABEL);
+	expect_co_query_access(arg, prop,
+			       DAOS_ACL_PERM_CONT_ALL & ~DAOS_ACL_PERM_GET_PROP,
+			       -DER_NO_PERM);
+	daos_prop_free(prop);
+
+	print_message("Non-access-related prop with only prop access\n");
+	prop = get_single_query_prop(DAOS_PROP_CO_LABEL);
+	expect_co_query_access(arg, prop,
+			       DAOS_ACL_PERM_GET_PROP,
+			       0);
+	daos_prop_free(prop);
+
+	/*
+	 * Ownership props can be accessed by users with either get-prop or
+	 * get-acl access
+	 */
+	print_message("Owner with only prop access\n");
+	prop = get_single_query_prop(DAOS_PROP_CO_OWNER);
+	expect_co_query_access(arg, prop,
+			       DAOS_ACL_PERM_GET_PROP,
+			       0);
+	daos_prop_free(prop);
+
+	print_message("Owner with only ACL access\n");
+	prop = get_single_query_prop(DAOS_PROP_CO_OWNER);
+	expect_co_query_access(arg, prop,
+			       DAOS_ACL_PERM_GET_ACL,
+			       0);
+	daos_prop_free(prop);
+
+	print_message("Owner with neither get-prop nor get-acl access\n");
+	prop = get_single_query_prop(DAOS_PROP_CO_OWNER);
+	expect_co_query_access(arg, prop,
+			       DAOS_ACL_PERM_CONT_ALL &
+			       ~(DAOS_ACL_PERM_GET_PROP |
+			         DAOS_ACL_PERM_GET_ACL),
```

Comment:
(style)  code indent should use tabs where possible

Reply:
Fixed